pkg/policy/selectorcache.go

```diff
@@ -695,132 +248,103 @@ func (l *labelIdentitySelector) fetchIdentityMappings() []identity.NumericIdenti
 // No locking needed.
 //
 
+// UpdateResult is a bitfield that indicates what parts of the policy engines
+// need to update in order to implement a policy
+type UpdateResult uint32
+
+const (
+	// UpdateResultUpdatePolicyMaps indicates the caller should call EndpointManager.UpdatePolicyMaps()
+	UpdateResultUpdatePolicyMaps = UpdateResult(1) << iota
+
+	// UpdateResultIdentitiesNeeded indicates the caller should wait for a complete ipcace label injection,
+	// as the selector is missing identities
+	UpdateResultIdentitiesNeeded
+
+	UpdateResultUnchanged = UpdateResult(0)
+)
+
 // UpdateFQDNSelector updates the mapping of fqdnKey (the FQDNSelector from a
-// policy rule as a string) to to the provided list of identities. If the contents
-// of the cachedSelections differ from those in the identities slice, all users
-// are notified asynchronously. Caller should Wait() on the returned
-// sync.WaitGroup before triggering any policy updates. Policy updates may need
-// Endpoint locks, so this Wait() can deadlock if the caller is holding any
-// endpoint locks.
-func (sc *SelectorCache) UpdateFQDNSelector(fqdnSelec api.FQDNSelector, identities []identity.NumericIdentity, wg *sync.WaitGroup) {
+// policy rule as a string) to to the provided list of IPs. If the contents
+// of the cachedSelections changes, all users are notified asynchronously.
+// Caller should Wait() on the returned sync.WaitGroup before triggering
+// any policy updates. Policy updates may need Endpoint locks, so this
+// Wait() can deadlock if the caller is holding any endpoint locks.
+//
+// Returns an UpdateResult that indicates if the caller should call UpdatePolicyMaps() and/or
+// wait for ipcache UpsertMetadata() to finish.
+func (sc *SelectorCache) UpdateFQDNSelector(fqdnSelec api.FQDNSelector, ips []netip.Addr, wg *sync.WaitGroup) UpdateResult {
```

Comment:
nit: This function comment is now incomplete, could do with an update. A few notes:

- This function will update the selector's mapping for IP addresses.
- It will also enqueue policymap updates for endpoints iff the IP is already used by another policy.
- This function will *not* initiate allocation of new identities, nor policymap updates for new identities, nor ipcache updates for new identities.
- Any IPs that were not previously allocated via another mechanism (such as another network policy or a previous DNS lookup) must be separately plumbed by callers into the ipcache, which will take care of ensuring updates to the datapath (including policymaps).